hail/src/main/scala/is/hail/io/LoadMatrix.scala

```diff
@@ -37,6 +35,8 @@ class LoadMatrixParser(rvb: RegionValueBuilder, fieldTypes: Array[Type], entryTy
              |    Line:
              |        ${ line.truncate }""".stripMargin
         )
+      off = addType(fieldTypes(ii))(line, rowNum, ii, off)
```

Comment:
Your test looks good. How does this change accomplish what you wanted? It looks like a simple reorder to me.

Reply:
The old way, we were checking that the line hadn't ended after every row field was added, which means that for something with no columns, we were checking that the line hadn't ended... at the end of the line.

This way, we're doing the check before adding each row field, so we'll only throw an error if we expect to be able to add another field and we can't. I think that's the only change necessary for this to work, but I wasn't able to get my dev env set up fully this week so I hadn't tested it locally.